### PR TITLE
Feature/comment resolving

### PR DESCRIPTION
**Pull request by jacobtoppm**
- Fixes an issue with deletions not being preserved when the page form is invalid and resubmitted
- Adds a new resolved status for comments

### CODE REVIEW

**Review by kaedroho — APPROVED** [review]

File: wagtail/core/models.py [hunk 1]
```diff
@@ -4885,6 +4884,8 @@ class Comment(ClusterableModel):
     revision_created = models.ForeignKey(PageRevision, on_delete=models.CASCADE, related_name='created_comments', null=True, blank=True)
     # Comments are only shown on revisions after revision_created
 
+    resolved = models.BooleanField(default=False)
```
[the remark this answers is not in the document]
Comment (jacobtoppm) [reply wagtail/core/models.py]: I've changed over in the latest commit, thanks!

File: client/src/components/CommentApp/components/Comment/index.tsx [hunk 2]
```diff
@@ -65,6 +65,14 @@ async function doDeleteComment(comment: Comment, store: Store) {
   }
 }
 
+async function resolveComment(comment: Comment, store: Store) {
```
[the remark this answers is not in the document]
Comment (jacobtoppm) [reply client/src/components/CommentApp/components/Comment/index.tsx]: Removed now, thanks!

**Review by kaedroho — APPROVED** [review]

**Comment by kaedroho** [comment]
There's a test failure on the `notz` environment, but other than that, LGTM!

**Comment by jacobtoppm** [comment]
> There's a test failure on the `notz` environment, but other than that, LGTM!

Thanks, just noticed! Should fix shortly, once the tests have passed I'll merge

**Pull request merged**